### PR TITLE
feat: cdn download tiny file

### PR DESCRIPTION
**Pull request by gaius-qi**
Signed-off-by: Gaius <gaius.qi@gmail.com>

## Description
- Support cdn download tiny file with range header

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: scheduler/resource/peer.go [hunk 1]
```diff
@@ -389,29 +394,35 @@ func (p *Peer) DeleteStream() {
...
 		Path:     fmt.Sprintf("download/%s/%s", p.Task.ID[:3], p.Task.ID),
 		RawQuery: "peerId=scheduler",
 	}
-	p.Log.Infof("download tiny file url: %#v", url)
 
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url.String(), nil)
 	if err != nil {
 		return []byte{}, err
 	}
 	req.Header.Set(headers.Range, fmt.Sprintf("bytes=%d-%d", 0, p.Task.ContentLength.Load()))
```
Comment (244372610) [inline scheduler/resource/peer.go:413]: why req need set range, if req has range header, The data downloaded by CDN has been processed by Range

Comment (244372610) [inline scheduler/resource/peer.go:413]: why req need set range, if req has range header, The data downloaded by CDN has been processed by Range

Comment (244372610) [inline scheduler/resource/peer.go:413]: why req need set range, if req has range header, The data downloaded by CDN has been processed by Range

Comment (gaius-qi) [reply scheduler/resource/peer.go:413]: Download according to the http protocol, direct request or range should be supported. You don't need to think about how to download on the other side, just support it, and it's not you to support it, but nginx to support it.

Comment (gaius-qi) [reply scheduler/resource/peer.go:413]: Why do you care what method the other uses to download? Does OSS care whether the download uses a range header or a direct request? I don't understand what you're thinking !

**Review by hyy0322 — APPROVED** [review]

File: scheduler/resource/peer.go (hunk 1 above)
Comment (244372610) [reply scheduler/resource/peer.go:413]: I think the scheduler should focus on scheduling and not care about the range of tasks

**Review by jim3ma — APPROVED** [review]
LGTM

**Pull request merged**